packages/fhir-router/src/repo.ts

```diff
@@ -457,10 +474,24 @@ export class MemoryRepository extends FhirRepository<undefined> {
     const { resourceType } = searchRequest;
     const resources = this.resources.get(resourceType) ?? new Map();
     const result = [];
-    for (const resource of resources.values()) {
-      if (matchesSearchRequest(resource, searchRequest)) {
-        result.push(resource);
+    try {
+      validateSearchResourceTypes(this, searchRequest);
+      if (resources.size === 0) {
+        // If there are no resources for this resource type,
+        // We still want to validate the search request and throw on any potentially invalid search params
+        // Instead of silently failing and returning an empty search set as if it was a valid search
+        matchesSearchRequest({ resourceType } as T, searchRequest);
       }
+      for (const resource of resources.values()) {
+        if (matchesSearchRequest(resource, searchRequest)) {
+          result.push(resource);
+        }
+      }
+    } catch (err) {
+      if (this.strictMode) {
```

Comment:
Added "strict mode" check to add an escape hatch when we don't want `MockClient` / `MemoryRepository` to throw, especially in the case of `storybook`